base/src/types/mod.rs

```diff
@@ -425,20 +442,19 @@ where
            serde(bound(serialize = "T: SerializeState<SeSeed>, Id: SerializeState<SeSeed>")))]
 pub struct AliasData<Id, T> {
     #[cfg_attr(feature = "serde_derive", serde(state))] pub name: Id,
-    /// Arguments to the alias
-    #[cfg_attr(feature = "serde_derive", serde(state))]
-    pub args: Vec<Generic<Id>>,
     /// The type that is being aliased
     #[cfg_attr(feature = "serde_derive", serde(state))]
     typ: T,
 }
 
-impl<Id, T> AliasData<Id, T> {
+impl<Id, T> AliasData<Id, T>
+where
+    T: From<Type<Id, T>>,
+{
     pub fn new(name: Id, args: Vec<Generic<Id>>, typ: T) -> AliasData<Id, T> {
```

Comment:
Wondering wif we could simplify this now by removing `args` from the constructor, and explicitly using `Type::forall` where necessary.